test/components/views/messages/RoomCreate-test.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+import { fireEvent, render, RenderResult } from "@testing-library/react";
+import { mocked } from "jest-mock";
+import { EventType, MatrixEvent } from "matrix-js-sdk/src/matrix";
+
+import dis from "../../../../src/dispatcher/dispatcher";
+import SettingsStore from "../../../../src/settings/SettingsStore";
+import RoomCreate from "../../../../src/components/views/messages/RoomCreate";
+import { stubClient } from "../../../test-utils/test-utils";
+import { Action } from "../../../../src/dispatcher/actions";
+
+jest.mock("../../../../src/dispatcher/dispatcher");
+
+describe("<RoomCreate />", () => {
+    const userId = "@alice:server.org";
+    const roomId = "!room:server.org";
+    const createEvent = new MatrixEvent({
+        type: EventType.RoomCreate,
+        sender: userId,
+        room_id: roomId,
+        content: {
+            predecessor: { room_id: "old_room_id", event_id: "tombstone_event_id" },
+        },
+        event_id: "$create",
+    });
+
+    function getComponent(event: MatrixEvent): RenderResult {
+        return render(<RoomCreate mxEvent={event} />);
+    }
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        mocked(dis.dispatch).mockReset();
+        jest.spyOn(SettingsStore, "getValue").mockReturnValue(false);
+        jest.spyOn(SettingsStore, "setValue").mockResolvedValue(undefined);
+        stubClient();
+    });
+
+    afterAll(() => {
+        jest.spyOn(SettingsStore, "getValue").mockRestore();
+        jest.spyOn(SettingsStore, "setValue").mockRestore();
+    });
+
+    it("Renders as expected", () => {
+        const wrapper = getComponent(createEvent);
+        expect(wrapper.asFragment()).toMatchSnapshot();
+    });
+
+    it("Links to the old version of the room", () => {
+        const wrapper = getComponent(createEvent);
+        expect(wrapper.getByText("Click here to see older messages.")).toHaveAttribute(
+            "href",
+            "https://matrix.to/#/old_room_id/tombstone_event_id",
+        );
+    });
+
+    it("Opens the old room on click", () => {
+        const wrapper = getComponent(createEvent);
+        const link = wrapper.getByText("Click here to see older messages.");
+        fireEvent.click(link);
```

Comment:
What would you think to using `userEvent` here instead? I'd say a larger gain to doing that than the previous suggestion, as userEvent fires a few more events (mouseover, hover etc.) than purely the click action so is a better model of user interaction. It would be imported and then used as `userEvent.click(link)` so would read the same.

Reply:
Sounds good - I was not aware of this - thanks!